ksqldb-tools/src/main/java/io/confluent/ksql/tools/migrations/commands/ApplyMigrationCommand.java

```diff
@@ -56,7 +77,188 @@
 
   @Override
   protected int command() {
-    throw new UnsupportedOperationException();
+    if (!validateConfigFilePresent()) {
+      return 1;
+    }
+
+    final MigrationConfig config;
+    try {
+      config = MigrationConfig.load(configFile);
+    } catch (KsqlException | MigrationException e) {
+      LOGGER.error(e.getMessage());
+      return 1;
+    }
+
+    return command(
+        config,
+        MigrationsUtil::getKsqlClient,
+        getMigrationsDirFromConfigFile(configFile),
+        Clock.systemDefaultZone()
+    );
+  }
+
+  @VisibleForTesting
+  int command(
+      final MigrationConfig config,
+      final Function<MigrationConfig, Client> clientSupplier,
+      final String migrationsDir,
+      final Clock clock
+  ) {
+    final Client ksqlClient;
+    try {
+      ksqlClient = clientSupplier.apply(config);
+    } catch (MigrationException e) {
+      LOGGER.error(e.getMessage());
+      return 1;
+    }
+
+    if (ValidateMigrationsCommand.validate(config, migrationsDir, ksqlClient)
+        && apply(config, ksqlClient, migrationsDir, clock)) {
+      ksqlClient.close();
+      return 0;
+    } else {
+      ksqlClient.close();
+      return 1;
+    }
+  }
+
+  private boolean apply(
+      final MigrationConfig config,
+      final Client ksqlClient,
+      final String migrationsDir,
+      final Clock clock
+  ) {
+    final int start = getStartVersion(config, ksqlClient);
+    final List<Migration> migrations;
+    LOGGER.info("Loading migration files");
+
+    if (next) {
+      migrations = getAllMigrations(start, start, migrationsDir);
+    } else if (version >= start) {
+      migrations = getAllMigrations(start, version, migrationsDir);;
+    } else if (version > 0 && version < start) {
+      LOGGER.error("Specified version is lower than latest migrated version");
+      return false;
+    } else {
+      migrations = getAllMigrations(start, migrationsDir);
+    }
```

Comment:
I think we should add another variant of the "apply" command that just applies a single, specific version. (We can use the "-v" flag.) Let's do this in a follow-up PR (I can take it up if you're tired of working on "apply" haha).